lib/chef/provider/aws_vpc_peering_connection.rb

```
@@ -0,0 +1,89 @@
+            }
+          ]
+        })
+      end
+      vpc_peering_connection
+    end
+  end
+
+  def update_aws_object(vpc_peering_connection)
+    vpc_id = vpc_peering_connection.requester_vpc_info.vpc_id
+    peer_vpc_id = vpc_peering_connection.accepter_vpc_info.vpc_id
+    peer_owner_id = vpc_peering_connection.accepter_vpc_info.owner_id
+
+    desired_vpc_id = Chef::Resource::AwsVpc.get_aws_object_id(new_resource.vpc, resource: new_resource)
+    desired_peer_vpc_id = Chef::Resource::AwsVpc.get_aws_object_id(new_resource.peer_vpc, resource: new_resource)
+    desired_peer_owner_id = new_resource.peer_owner_id
+
+    if vpc_id != desired_vpc_id
+      raise "VCP peering connection requester vpc cannot be changed after being created! Desired requester vpc id for #{new_resource.name} (#{vpc_peering_connection.id}) was \"#{desired_vpc_id}\" and actual id is \"#{vpc_id}\""
+    end
+    if peer_vpc_id != desired_peer_vpc_id
+      raise "VCP peering connection accepter vpc cannot be changed after being created! Desired accepter vpc id for #{new_resource.name} (#{vpc_peering_connection.id}) was \"#{desired_peer_vpc_id}\" and actual id is \"#{peer_vpc_id}\""
+    end
+    unless new_resource.peer_owner_id.nil?
+      if peer_owner_id !=  desired_peer_owner_id
+        raise "VCP peering connection accepter owner id vpc cannot be changed after being created! Desired accepter vpc owner id for #{new_resource.name} (#{vpc_peering_connection.id}) was \"#{desired_peer_owner_id}\" and actual owner id is \"#{peer_owner_id}\""
+      end
+    end
+
+    update_status(vpc_peering_connection, new_resource)
+  end
+
+  def destroy_aws_object(vpc_peering_connection)
+    converge_by "delete #{new_resource.to_s} in #{region}" do
+      begin
+        unless ['deleted', 'failed', 'deleting'].include? vpc_peering_connection.status.code
+          vpc_peering_connection.delete
+        end
+      rescue Aws::EC2::Errors
+        raise "#{new_resource.to_s} could not be deleted."
```

Yup, let it bubble up!